app/models/tenant.rb

```diff
@@ -240,6 +240,12 @@ def any_editable_domains?
     ae_domains.where(:system => false).count > 0
   end
 
+  def reset_domain_priority_by_ordered_ids(ids)
+    uneditable_domains = visible_domains - editable_domains
+    uneditable_domains.delete_if { |domain| domain.name == 'ManageIQ' }
```

Comment:
Can this logic be pushed down into the MiqAeDomain model?  Also should be using `MiqAeDatastore::MANAGEIQ_DOMAIN` instead of 'ManageIQ'.